src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -0,0 +1,204 @@
+                using var asyncToken = _asyncListener.BeginAsyncOperation(GetType() + ".Search");
+                _progress.AddItems(_solution.Projects.Count());
+
+                var workspace = _solution.Workspace;
+
+                // If the workspace is tracking documents, use that to prioritize our search
+                // order.  That way we provide results for the documents the user is working
+                // on faster than the rest of the solution.
+                var docTrackingService = workspace.Services.GetService<IDocumentTrackingService>();
+                if (docTrackingService != null)
+                {
+                    await SearchProjectsInPriorityOrderAsync(docTrackingService).ConfigureAwait(false);
+                }
+                else
+                {
+                    await SearchAllProjectsAsync().ConfigureAwait(false);
+                }
+            }
+            catch (OperationCanceledException)
+            {
+            }
+            finally
+            {
+                var service = _solution.Workspace.Services.GetRequiredService<IWorkspaceStatusService>();
+                var isFullyLoaded = await service.IsFullyLoadedAsync(_cancellationToken).ConfigureAwait(false);
+                // providing this extra information will make UI to show indication to users
+                // that result might not contain full data
+                _callback.Done(isFullyLoaded);
+            }
+        }
+
+        private async Task SearchProjectsInPriorityOrderAsync(IDocumentTrackingService docTrackingService)
+        {
+            var processedProjects = new HashSet<Project>();
+
+            var activeDocument = docTrackingService.GetActiveDocument(_solution);
+            var visibleDocs = docTrackingService.GetVisibleDocuments(_solution)
+                                                .Where(d => d != activeDocument)
+                                                .ToImmutableArray();
+
+            // First, if there's an active document, search that project first, prioritizing
+            // that active document and all visible documents from it.
+            if (activeDocument != null)
+            {
+                var activeProject = activeDocument.Project;
+                processedProjects.Add(activeProject);
+
+                var visibleDocsFromProject = visibleDocs.Where(d => d.Project == activeProject);
+                var priorityDocs = ImmutableArray.Create(activeDocument).AddRange(visibleDocsFromProject);
+
+                // Search the active project first.  That way we can deliver results that are
+                // closer in scope to the user quicker without forcing them to do something like
+                // NavToInCurrentDoc
+                await Task.Run(() => SearchAsync(activeProject, priorityDocs), _cancellationToken).ConfigureAwait(false);
```

Comment:
Is this new code, or old code? 

It seems odd to me that we await here. The aim is to provide relevant results quicker, which makes sense, but we're also delaying other results? Or is it just that the actual searching would become the bottleneck if this wasn't awaited?

Reply:
old, just moved from here - http://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures.Wpf/NavigateTo/NavigateToItemProvider.Searcher.cs,129

Since I'm not sure, unless there is a strong reason to change I'll leave as-is.